jina/parsers/orchestrate/deployment.py

```diff
@@ -43,6 +43,14 @@ def mixin_base_deployment_parser(parser):
         'This Deployment will not be context managed by the Flow.',
     )
 
+    gp.add_argument(
+        '--grpc-metadata',
+        action=KVAppendAction,
+        metavar='KEY: VALUE',
+        nargs='*',
+        help='The metadata to be passed to the gRPC request. This argument only applies for external Deployments.',
```

Comment:
why only affects external Deployments?

Reply:
For now, the metadata is never used by the internal/normal executor deployment. I cannot see the use cases except expertnal executor.

Reply:
it shluld be generic

Reply:
In the latest commit, the metadata can be applied to any deployment, it's up to the user's decision.